spec.md

```diff
@@ -615,6 +621,24 @@ message PublishVolumeInfo {
   // possible.
   map<string, string> values = 1;
 }
+
+message Principal {
+  // describes attributes for defining the principal identity of
+  // a volume binding for a container.
+
+  // This field is OPTIONAL. If specified it contains the CO namespace
+  // identity in which the the following identity information (user /
+  // group) exists.
+  string namespace = 1;
+  // This field is OPTIONAL. If specified it contains the CO user
+  // identity holding ownership of a volume, similar to the user_id
+  // in a POSIX file system.
+  string user = 2;
+  // This field is OPTIONAL. If specified it contains the CO group
+  // identities holding ownership of a volume, similar to the group_id
+  // in a POSIX file system.
+  repeated string group = 3;
+}
```

Comment:
It looks like this is trying to establish some ACLs. The "single user, multi group" feels a bit too constraining. Many systems support more flexible models than this. Perhaps we should model `Principal` a bit more along the lines of:
```proto
message Principal {
  message User {
    string name
  }
  message Group {
    string name
  }
  message Role {
    string name
  }
  message Everyone {}

  string namespace

  oneof value {
    User user
    Group group
    Role role
    Everyone everyone
  }
}
```
...and then make it a `repeated` field in the publish calls?

It's also weird because the publish calls assume that all parties will be granted the same "access mode".

Perhaps we should remodel the access/accessed_as fields as a list of tuples?
```go
message PrincipalAccess {
  // both fields are REQUIRED
  bool readonly
  Principal principal
}
message ControllerPublishVolumeRequest {
  ...
  // replace readonly and accessed_as with the following:
  // REQUIRED. If not interested in user/group/role or namespace, then specify
  // a single PrincipalAccess { readonly = ..., principal = { everyone = {} } }
  repeated PrincipalAccess principal_access
}
```